src/Console/Database/FixHtmlEncodingCommand.php

```diff
@@ -97,54 +144,51 @@ protected function configure()
             __('Itemtype to fix')
         );
 
-        $this->addOption(
-            'id',
-            null,
-            InputOption::VALUE_IS_ARRAY | InputOption::VALUE_REQUIRED,
-            __('Id of the item to fix')
-        );
-
-        $this->addOption(
-            'field',
-            null,
-            InputOption::VALUE_IS_ARRAY | InputOption::VALUE_REQUIRED,
-            __('Field of the item to fix')
-        );
-
         $this->addOption(
             'dump',
             null,
             InputOption::VALUE_OPTIONAL,
             __('Path of file containing dump of existing values.')
         );
 
-        $this->addUsage('--itemtype=ITILFollowup --id=42 --field=content [--dump=file_path.sql]');
+        $this->addUsage('--itemtype=ITILFollowup [--dump=file_path.sql]');
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        global $DB;
-
         $this->checkArguments();
+        $this->findTextFields();
+        $this->scanItems();
+
+        $count = $this->countItems($this->invalid_items);
+        if ($count === 0) {
+            $output->writeln('<info>' . __('No invalid item found.') . '</info>');
+            return 0;
+        }
 
-        $itemtype = $input->getOption('itemtype');
+        $output->writeln('<info>' . sprintf(_n('%d invalid item found.', '%d invalid items found.', $count), $count) . '</info>');
+        $this->askForConfirmation();
 
         if ($input->getOption('dump')) {
             $this->dumpObjects();
         }
 
+        $this->confirm = $this->askForItemConfirmation();
+
         $this->fixItems();
 
-        if (count($this->failed_items) > 0) {
+        if ($this->countItems($this->failed_items) > 0) {
             $this->output->writeln(
                 '<error>' . sprintf(__('Unable to update %s items'), count($this->failed_items)) . '</error>',
                 OutputInterface::VERBOSITY_QUIET
             );
-            foreach ($this->failed_items as $item_id) {
-                $this->output->writeln(
-                    '<error>' . sprintf(__('Itemtype %s ID %s'), $itemtype, $item_id) . '</error>',
-                    OutputInterface::VERBOSITY_QUIET
-                );
+            foreach ($this->failed_items as $itemtype) {
+                foreach ($this->failed_items as $item_id => $item) {
```

Comment:
There is probably a mistake here. There is a loop on `$this->failed_items` inside a loop on same array.

Reply:
```suggestion
            foreach ($this->failed_items as $itemtype => $items) {
                foreach ($items as $item_id => $item) {
```